src/components/ImagePicker/index.js

```diff
@@ -177,69 +179,134 @@ export default React.createClass({
-                multiple={this.props.multiple}
-                name={this.props.name}
-                onDrop={this.onDrop}
-                onDropAccepted={this.onDropAccepted}
-                onDropRejected={this.onDropRejected}
+                htmlFor="image-picker--file-input"
               >
-                <div className="input--tile-input-description">
-                  Select or drag an image relative to the shopping category
-                </div>
-                <div className="input--tile-input-icon">
-                  <Icon type="plus" size="regular" />
-                </div>
-              </Dropzone>
-            </label>
-            {!this.props.disableUrlInput ? (
-              <InputImageUrl
-                onInputChange={this.props.onUrlInputChange}
-                onSubmit={this.onUrlSubmit}
-                placeholder={this.props.urlPlaceholder}
-              />
-            ) : null}
-          </div>
+                {!this.props.disablePreviewImage &&
+                  this.state.previewImage &&
+                  !(this.props.fileError || this.props.urlError || this.props.error) ? (
+                    // eslint-disable-next-line
+                    <div className="image-picker--preview-image-container">
+                      <img
+                        className="image-picker--preview-image"
+                        src={this.state.previewImage.url}
+                        alt={this.state.previewImage.alt}
+                        style={{
+                          width: '100%',
+                          height: 'auto',
+                        }}
+                      />
+                    </div>
+                  ) : null}
+                <Dropzone
+                  accept={this.props.accept}
+                  className={classNames(
+                      'content',
+                      'dropzone',
+                      { 'dropzone_on-drag': this.state.isDragging },
+                  )}
+                  disableClick={this.props.disableClick}
+                  inputProps={this.props.inputProps}
+                  maxSize={this.props.maxSize}
+                  minSize={this.props.minSize}
+                  multiple={this.props.multiple}
+                  name={this.props.name}
+                  onDrop={this.onDrop}
+                  onDropAccepted={this.onDropAccepted}
+                  onDropRejected={this.onDropRejected}
+                >
+                  <div className="image-picker--tile-input-description">
+                    Select or drag an image relative to the shopping category
+                  </div>
+                  <div className="image-picker--tile-input-icon">
+                    <Icon type="plus" size="regular" />
+                  </div>
+                </Dropzone>
+              </label>
+
+              {(((!this.state.previewImage ||
+                (this.state.previewImage && this.disablePreviewImage)) &&
```

Comment:
I'll see what I can do.  I am open for suggestions on how this can be better implemented.